File: rust/src/models/revocation_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

/// Token revocation request
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct RevocationRequest {
    /// The token to revoke
    pub token: String,

    /// Optional hint about the token type (access_token or refresh_token)
    #[serde(skip_serializing_if = "Option::is_none")]
    pub token_type_hint: Option<String>,
}

impl RevocationRequest {
    pub fn new(token: String) -> Self {
        Self {
            token,
            token_type_hint: None,
        }
    }
}

File: typescript/src/models/revocationRequest.ts
// this file is @generated
/** Token revocation request */
export interface RevocationRequest {
  /** The token to revoke */
  token: string;

  /** Optional hint about the token type (access_token or refresh_token) */
  tokenTypeHint?: string | null;
}

export const RevocationRequestSerializer = {
  _fromJsonObject(object: any): RevocationRequest {
    return {
      token: object["token"],
      tokenTypeHint: object["token_type_hint"],
    };
  },

  _toJsonObject(self: RevocationRequest): any {
    return {
      token: self.token,
      token_type_hint: self.tokenTypeHint,
    };
  },
};
